db/upstream.go

```diff
@@ -4,71 +4,189 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/flanksource/duty/models"
 	"github.com/flanksource/incident-commander/api"
 	"gorm.io/gorm/clause"
 )
 
+// TODO: Is it ncessesary to check for agent_id IS NULL here?
+// At this point we're sure that it's an agent so the agent_id is always going to be nil UUID.
+// Unless this is both an agent and an upstream ... ?
+func GetAllMissingResourceIDs(ctx context.Context, req *api.PushedResourceIDs) (*api.PushData, error) {
+	var upstreamMsg api.PushData
+
+	if err := Gorm.WithContext(ctx).Not(req.Components).Find(&upstreamMsg.Components).Error; err != nil {
+		return nil, fmt.Errorf("error fetching components: %w", err)
+	}
+
+	if err := Gorm.WithContext(ctx).Not(req.ConfigItems).Find(&upstreamMsg.ConfigItems).Error; err != nil {
+		return nil, fmt.Errorf("error fetching config items: %w", err)
+	}
+
+	if err := Gorm.WithContext(ctx).Not(req.Canaries).Find(&upstreamMsg.Canaries).Error; err != nil {
+		return nil, fmt.Errorf("error fetching canaries: %w", err)
+	}
+
+	if err := Gorm.WithContext(ctx).Not(req.Checks).Find(&upstreamMsg.Checks).Error; err != nil {
+		return nil, fmt.Errorf("error fetching checks: %w", err)
+	}
+
+	if err := Gorm.WithContext(ctx).Not(req.ConfigAnalysis).Find(&upstreamMsg.ConfigAnalysis).Error; err != nil {
+		return nil, fmt.Errorf("error fetching config analyses: %w", err)
+	}
+
+	if err := Gorm.WithContext(ctx).Not(req.ConfigChanges).Find(&upstreamMsg.ConfigChanges).Error; err != nil {
+		return nil, fmt.Errorf("error fetching config changes: %w", err)
+	}
+
+	checkStatusQuery := Gorm.WithContext(ctx)
+	for _, cs := range req.CheckStatuses {
```

Comment:
This will not perform very well as there will be a huge number of statuses - I think analysis, changes, relationships, can also be skipped - they will retry after the config item and check have reconciled